blog/_posts/2017-11-27-view-based-collections.md

```diff
@@ -0,0 +1,163 @@
+---
+layout: blog-detail
+post-type: blog
+by: Julien Richard-Foy
+title: Let Them Be Lazy!
+---
+
+One of the main changes in the new design of the collections is that operations have
+“view based” default implementations rather than “builder based”. This blog article
+explains what this means as well as what it means for a collection to be lazy or
+strict, how the view based design allows us to support both types of collections,
+and finally why this is an improvement over the current design of the collections.
+
+More background information about the collections redesign can be found in
+[this blog post](http://www.scala-lang.org/blog/2017/02/28/collections-rework.html). The
+source code is available in
+[this GitHub repository](https://github.com/scala/collection-strawman).
+
+## Background: Lazy vs Strict Collections
+
+We can classify collections in two categories: those that evaluate their elements when
+they are created, and those that evaluate their elements when they are needed only.
+
+An example of the first category is `List[A]`: when we create a list we compute all
+its elements, when we transform a list (e.g. using `map`) we compute all the elements
+of the resulting list. We say that list is a **strict** collection type.
+
+An example of the second category is `Stream[A]`: we can create and transform streams
+without computing their elements. Those might actually never be computed unless we iterate
+through them. We say that streams are **lazy**.
+
+Laziness allows us to create fancy infinite collections without blowing the memory:
+`Stream.continually(42)`. In practice, lazy collections are particularly useful to
+describe successive transformation operations without evaluating intermediate
+transformations. Typically, if we want to transform the first `n` elements of
+a list using `map` we will probably write something like `xs.take(n).map(f)`.
+Unfortunately, this expression will create an intermediate collection (resulting from
+the `.take(n)` call) although we are only interested in the result of the `map` call.
+Creating this intermediate collection can be an expensive operation that we can
+avoid by using a view, which is a lazy collection: `xs.view.take(n).map(f)`.
+
+## What’s Wrong With Views?
+
+Unfortunately, the current support of views in the standard library suffers from
+two issues.
+
+### Operations Default Implementations Break Laziness
+
+In the current collections the default implementations of transformation operations
+are based on strict builders. Consequently, when they are inherited by lazy collections
```

Comment:
what are "they" and "these operations"?

I got a bit lost in this paragraph, it might help with a basic example